sqlmesh/core/engine_adapter/base.py

```diff
@@ -1400,30 +1410,42 @@ def insert_append(
             target_table=table_name,
             source_columns=source_columns,
         )
-        self._insert_append_source_queries(table_name, source_queries, target_columns_to_types)
+        self._insert_append_source_queries(
+            table_name, source_queries, target_columns_to_types, track_execution_stats
+        )
 
     def _insert_append_source_queries(
         self,
         table_name: TableName,
         source_queries: t.List[SourceQuery],
         target_columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        track_execution_stats: bool = True,
     ) -> None:
         with self.transaction(condition=len(source_queries) > 0):
             target_columns_to_types = target_columns_to_types or self.columns(table_name)
             for source_query in source_queries:
                 with source_query as query:
-                    self._insert_append_query(table_name, query, target_columns_to_types)
+                    self._insert_append_query(
+                        table_name,
+                        query,
+                        target_columns_to_types,
+                        track_execution_stats=track_execution_stats,
+                    )
 
     def _insert_append_query(
         self,
         table_name: TableName,
         query: Query,
         target_columns_to_types: t.Dict[str, exp.DataType],
         order_projections: bool = True,
+        track_execution_stats: bool = True,
```

Comment:
Until we know what other execution stats we need to track, can we maybe be more specific with the name?